app/src/main/java/com/jerboa/ui/components/home/HomeViewModel.kt

```diff
@@ -42,7 +44,8 @@ class HomeViewModel : ViewModel() {
     private var blockPersonRes: ApiState<BlockPersonResponse> by mutableStateOf(ApiState.Empty)
     var fetchingMore by mutableStateOf(false)
         private set
-
+    var refreshing by mutableStateOf(false)
```

Comment:
The refreshindicator should only show if you pull down, every other action that causes it to load the posts (such as login, the more options refresh button) also trigger the ApiState.loading thus i cant use that, so I added a bunch of refreshing states that only track the state of when it is triggered by the user pulling down the indicator

Reply:
But getPosts and the others then set it back to loading

Reply:
Then you could make another viewmodel function that doesn't set loading, or make it optional in some way.

Either way it makes sense to have a single ApiState object reflect what's going on, rather than multiple approaches on top of each other.

Reply:
I'm not following with `that makes loading optional`. I'm just doing a approach that's already been done, it's same as fetchingMore. Loading is still used for the loading bar. Adding a Apistate imo is much more complexity because now each exhaustive "when" structure will have to be updated to have this state that's completely irrelevant for those causes. To the api refreshing is irrelevant it doesn't need to differentiate between refreshing an action that's only done when a user pulls down an indicator vs when its waiting for a result. I can understand the onCompletion callback if you have another way, I'm happy to change it. But i don't understand why against the refreshing state. The viewmodels are littered with states already.

Reply:
I really hated added fetchingMore, but it was unfortunately necessary for infinite scrolling. Ideally there shouldn't be any states besides ApiState.

I spose I'm okay though with this refreshing state, although I see this getting complicated very quickly with multiple loading and conflicting Api states.

Reply:
I think its actually possible to remove fetchingMore, and I see issue with it too that I can fix. If fetching more fails it will still skip to the next page and on next try it will skip loading that page and loads the next.

Reply:
I too noticed that it's skipping pages when failing. How about making `fetchingMore` an ApiState of the current page and `postRes` just plain list? The `postRes` can be used directly in the lazy list without matching. The `fetchingMore` can be used to show loading state & possibly retry on error. If `fetchingMore` is loading & page is 1, it's refreshing and `postRes` has to be reset. Otherwise it's just loading and `postRes` will be extended with the result. Also, when incrementing pages if the previous page ended in failure (ie `fetchingMore` is ApiState.Failure) then retry the page instead. However, I think this might be a lot more work.

Reply:
I have already a working change that removes fetchingmore, ` loading & page is 1` this still doesn't guarantee that it only shows the indicator when you pull it down, for example you can use the refresh button in the more options on the first page and then it shouldnt show the indicator. And i have already included the fix fort he failing pages ` then retry the page instead` exactly what I am doing